pkg/target/factory/factory.go

```diff
@@ -827,6 +870,14 @@ func (f *TargetFactory) mapSecretValues(config any, ref, mountedSecret string) {
 			c.Config.Channel = values.Channel
 		}
 
+	case *v1alpha1.Config[v1alpha1.SplunkOptions]:
+		if values.Webhook != "" {
```

Comment:
You are checking for Webhook and setting Host, you should adopt test cases for splunk to ensure its working